src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDag.cs

```diff
@@ -39,9 +39,16 @@ public HashSet<LabelSymbol> ReachableLabels
             {
                 if (_reachableLabels == null)
                 {
-                    // PROTOTYPE(patterns2): avoid linq?
-                    _reachableLabels = new HashSet<LabelSymbol>(
-                        this.TopologicallySortedNodes.OfType<BoundLeafDecisionDagNode>().Select(d => d.Label));
+                    var result = new HashSet<LabelSymbol>();
```

Comment:
Consider if `SmallDictionary` would be more appropriate here. I believe it has significant size and speed advantages at < 30 elements.

Reply:
`SmallDictionary` has the wrong API for the clients of this API.

---
In reply to: [194895435](https://github.com/dotnet/roslyn/pull/27753#discussion_r194895435) [](ancestors = 194895435)